File: problems/problems_3740/solution.ts
function minimumDistance(nums: number[]): number {
    let ans = Number.MAX_SAFE_INTEGER;
    const record: Map<number, number[]> = new Map();
    for (let k = 0; k < nums.length; k++) {
        const num = nums[k];
        if (!record.has(num)) {
            record.set(num, []);
        }
        record.get(num)!.push(k);
        const indices = record.get(num)!;
        if (indices.length > 2) {
            const i = indices[indices.length - 3];
            ans = Math.min(ans, (k - i) * 2);
        }
    }
    return ans === Number.MAX_SAFE_INTEGER ? -1 : ans;
};

export function Solve(inputJsonElement: string): any {
	const inputValues: string[] = inputJsonElement.split("\n");
	const nums: number[] = JSON.parse(inputValues[0]);
	return minimumDistance(nums);
}

File: problems/problems_3740/solution.rs
use serde_json::{json, Value};

pub struct Solution;

impl Solution {
    pub fn minimum_distance(nums: Vec<i32>) -> i32 {
        use std::collections::HashMap;
        let mut ans = i32::MAX;
        let mut record: HashMap<i32, Vec<usize>> = HashMap::new();
        for (k, &num) in nums.iter().enumerate() {
            record.entry(num).or_insert_with(Vec::new).push(k);
            if let Some(indices) = record.get(&num) {
                if indices.len() > 2 {
                    let i = indices[indices.len() - 3];
                    ans = ans.min(((k - i) * 2) as i32);
                }
            }
        }
        if ans == i32::MAX { -1 } else { ans }
    }
}

#[cfg(feature = "solution_3740")]
pub fn solve(input_string: String) -> Value {
	let input_values: Vec<String> = input_string.split('\n').map(|x| x.to_string()).collect();
	let nums: Vec<i32> = serde_json::from_str(&input_values[0]).expect("Failed to parse input");
	json!(Solution::minimum_distance(nums))
}
